source/agora/node/Ledger.d

```diff
@@ -2178,7 +2194,45 @@ unittest
     scope ledger = new TestLedger(genesis_validator_keys[0], blocks, params, 600.seconds, mock_clock);
 
     // Add preimages for all validators (except for two of them) till end of cycle
-    auto skip_indexes = [ 2, 5 ];
+    uint[] skip_indexes = [ 2, 5 ];
+
+    ledger.simulatePreimages(Height(params.ValidatorCycle), skip_indexes);
+
+    // Block with no fee
+    auto no_fee_txs = blocks[$-1].spendable.map!(txb => txb.sign()).array();
+    no_fee_txs.each!(tx => assert(ledger.acceptTransaction(tx)));
+
+    ConsensusData data;
+    ledger.prepareNominatingSet(data, Block.TxsInTestBlock, mock_clock.networkTime());
+
+    assert(ledger.validateConsensusData(data, skip_indexes) is null);
+
+    data.missing_validators = [2,5,7];
+    assert(ledger.validateConsensusData(data, [2,5,7,9]) is null);
```

Comment:
```suggestion
    assert(ledger.validateConsensusData(data, [2,3,5,7,9]) is null);
```

Reply:
Not sure this will pass but I think it should.

Reply:
here we are testing: 
[2,5] ⊆ [2,5,7] ⊆ [2,5,7,9] , so of course your suggestion would also pass

[2,5] ⊆ [2,5,7] ⊆ [2,3,5,7,9], but I think no need to change the code

Reply:
> here we are testing:
> [2,5] ⊆ [2,5,7] ⊆ [2,5,7,9] , so of course your suggestion would also pass
> 
> [2,5] ⊆ [2,5,7] ⊆ [2,3,5,7,9], but I think no need to change the code

Are you sure? The reason why I suggested the change is I suspected it would fail with the current implementation.

Reply:
I just checked locally and the test fails with the change. The reason being that the `startsWith` is not same as sub set check.

Reply:
sorry, I misunderstood, I thought you have a problem with the test. Yes the implementation of the subset had a bug, thanks.